doc/specs/#182 - Support for installation of portable standalone apps.md

```diff
@@ -0,0 +1,190 @@
+
+### Command Line Arguments
+
+Users should be able to specify the location for where the program is “installed” on their machine with the “--location” argument.
+
+Some packages like GitLab Runner have a file name including extra metadata (gitlab-runner-windows-386.exe). The installation instructions suggest renaming the file after it has been downloaded. In that vein, portable / standalone executables should have their “command” value specified so the Windows Package Manager can perform a default rename behavior. In addition, a “--rename” argument should be added so the user can choose a value they prefer. To have the upgrade scenario honor the custom name, this information should be recorded in the installed packages data store.
+
+Some portable applications generate or consume other files. An additional argument "--purge" will be added for the uninstall scenario to remove all files and subsequently the directory if the user wishes. The corollary argument "--preserve" will be used to preserve files if the default setting has been modified.
+
+### Settings
+
+Users should have settings to be able to specify a new default location other than the system default for the Windows Package Manager.
+
+The default path for installing these packages is "%LOCALAPPDATA%/WinGet_Packages/User" for user based installs. The default path for installing these packages is "%LOCALAPPDATA%/WinGet_Packages/Machine"for machine wide installs.
+
+The corresponding settings are "PortablePackageUserRoot" and "PortablePackageUserRoot".
+
+>Note: The "packageIdentifier" will be used to generate subdirectories for these binaries to be installed to.
+
+Using Gitlab.gitlab runner as an example would result in the .exe being placed in "%LOCALAPPDATA%/WinGet_Packages/User/Gitlab.gitlab-runner/"
+If a user has configured a path to be used for portable applications, that path should be honored. The user should also be able to specify if they want all portable packages placed in the same directory, or in a directory per package.
+
+A related setting for the uninstall scenario will be able to specify the default behavior for either "--clean" or "--purge".
+
+### Install
+
+When the portable application is being installed by the Windows Package Manager, an entry will be created in Windows Apps & Features so the user will be able to see that the application is installed.
+
+The "AppPath" registry subkey will identify the path to the application so the user will be able to execute the program from any path via command prompt.
+
+The data from “AppsAndFeaturesEntry” in the manifest will be used to specify the name of the package in Windows Apps & Features.  Constraints for manifests with portable packages are covered below in the section on validating manifests. If the “AppsAndFeaturesEntry” doesn’t have a “DisplayName” value, then we will use the “PackageName”.
+The Product Code will be populated with the "PackageIdentifier" if it is not specified in the “AppsAndFeaturesEntry”.
```

Comment:
I think that it would be useful to have the Product Code also be tied to a specific installer/version like it is with MSI installers. Perhaps it could be to the SHA256, or the "PackageVersion"?

Reply:
That's a good point.

Reply:
Product Codes that change over time are actually quite annoying.  I would prefer that we don't do that.

Reply:
> Product Codes that change over time are actually quite annoying. I would prefer that we don't do that.

As long as there aren't any potential issues with matching a program to an ARP entry, or disambiguation, then I would think that having static product codes would be okay. I would assume that the DisplayName and DisplayVersion would always be set, so it may not be an issue unless it conflicts with an already installed application making the package ambiguous